servlet/src/main/java/com/linecorp/armeria/server/servlet/UrlMapper.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.servlet;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+import javax.annotation.Nullable;
+
+/**
+ * Url mapping
+ * Mapping specification
+ * In the web application deployment descriptor.
+ */
+final class UrlMapper<T> {
+    private final boolean singlePattern;
+    private final List<Element<T>> elementList = new ArrayList<>();
+    private final StringUtil stringUtil = new StringUtil();
+
+    /**
+     * Creates a new instance.
+     */
+    UrlMapper(boolean singlePattern) {
+        this.singlePattern = singlePattern;
+    }
+
+    /**
+     * Add mapping.
+     */
+    void addMapping(String urlPattern, T object, String objectName) {
+        requireNonNull(urlPattern, "urlPattern");
+        requireNonNull(object, "object");
+        requireNonNull(objectName, "objectName");
+        if (elementList.stream()
+                       .filter(x -> singlePattern && x.name.equals(objectName))
+                       .findAny().orElse(null) != null) {
+            throw new IllegalArgumentException("The [" + objectName + "] mapping exist!");
+        }
+
+        final Element element = elementList.stream()
+                                           .filter(x -> x.pattern.equals(urlPattern))
+                                           .findFirst().orElse(null);
+        if (element != null) {
+            element.name = objectName;
+            element.object = object;
+        } else {
+            elementList.add(new Element<>(urlPattern, object, objectName));
+        }
+    }
+
+    /**
+     * Gets a mapping object.
+     */
+    @Nullable
+    Element<T> getMapping(String absoluteUri) {
+        requireNonNull(absoluteUri, "absoluteUri");
+        return elementList.stream()
+                          .filter(x -> stringUtil.match(x.pattern, absoluteUri, "*"))
```

Comment:
The `allToken` is always `*` so we do not want to specify it.
The class `StringUtil` is a bit complex because it's for ANT mapping. How about just using `Pattern` for simplicity?
We can just consider three cases:
- The path that ends with `*`. (e.g `/foo/bar/*`)
- Implicit mapping. (e.g `*.bop`)
- Exact path matching.

We might revisit this to optimize later though.

Reply:
Yes, I think so. We will revisit this to optimize later.

Reply:
What I meant is that I think we need to remove the `StringUtil` class and just use `Pattern` in this PR. And then We revisit this to optimize it. 😆

Reply:
Oh, I understand now :D. I will remove StringUtil.